src/main/webapp/script.js

```diff
@@ -22,3 +21,129 @@ function getDatabases(){
   });
 }
 
+function submitDatabaseForm() { 
+  document.getElementById("database-select-form").submit();
+}
+
+//Get database and selected tables from query string and /tables-from-db
+function getDatabaseAndTable(){
+    const tablesUrl = '/tables-from-db';
+    const search = window.location.search;
+    const queryString = tablesUrl + search;
+    var startIndex = 0;
+    var databaseString= "";
+    
+    for (var i = 0; i<search.length; i++){
+        if (search.charAt(i) == '='){
+            startIndex = i;
+        }
+        else if (search.charAt(i) == '&'){
+            break;
+        }
+        else{
+             databaseString += search.charAt(i);
+        }
+    }
+
+    databaseString = databaseString.substring(startIndex);
+    fetch(queryString)
```

Comment:
This way definitely works! I'm not sure you even need to do a fetch here, since the database and tables are already in the URL of this page. As you got the database name from the search string, maybe you could also get the selected tables from there rather than doing a fetch?

Reply:
That might improve running time so that is a good idea!